src/algorithm/good_thomas_algorithm.rs

```diff
@@ -9,26 +11,22 @@ use algorithm::FFTAlgorithm;
 pub struct GoodThomasAlgorithm<T> {
     width: usize,
     // width_inverse: usize,
-    width_size_fft: Box<FFTAlgorithm<T>>,
+    width_size_fft: Rc<FFTAlgorithm<T>>,
 
     height: usize,
     // height_inverse: usize,
-    height_size_fft: Box<FFTAlgorithm<T>>,
+    height_size_fft: Rc<FFTAlgorithm<T>>,
 
     input_map: Vec<usize>,
     output_map: Vec<usize>,
-
-    scratch: Vec<Complex<T>>,
 }
 
-impl<T> GoodThomasAlgorithm<T>
-    where T: Signed + FromPrimitive + Copy
-{
-    pub fn new(n1: usize,
-               n1_fft: Box<FFTAlgorithm<T>>,
-               n2: usize,
-               n2_fft: Box<FFTAlgorithm<T>>)
-               -> Self {
+impl<T: FFTnum> GoodThomasAlgorithm<T> {
+    #[allow(dead_code)]
```

Comment:
The entire algorithm. It currently isn't used by the planner, because the MixedRadix struct is faster. I kept it around because it's a useful reference implementation for implementing butterflies. For example, I used it to debug my size-6 butterfly, implementation, which uses the good-thomas algorithm.

It might be best to get rid of it if it's not being used. Or maybe put it behind a #[cfg(test)]? what do you think?